### PR TITLE
cook steak with welding tools like a boss

### PR DESCRIPTION
**Pull request by ghost**
## What this does

This is a draft PR aimed to address #35122 and enable us to cook simple food items like meat with welding tools, lighters, etc. I think I've figured out an okay way to do this but there remain things to be done. Copy pasta-ing my post on the issue:

1. Make the cooking time dependent on the existent cooking time with a microwave and the energy the heating element can apply (my code makes this cookable by any item with an is_hot proc returning true).
2. We need a "cooked meat" sprite and item. Currently there is no recipe accepting just one piece of meat, I had to just remove the salt and pepper requirement from the Meat steak recipe to test. We need to go through the [recipes_microwave.dm](https://github.com/vgstation-coders/vgstation13/blob/24d0c3327d1cc605340f8bdc8f19c897a3889a60/code/modules/food/recipes_microwave.dm) file to see if there are even any items that can be cooked without reagents, and we need to add a "cooked meat" or "bland meat" recipe and file to allow people to actually cook the various kinds of meat like normal meat, monkey meat, synthmeat, xeno meat etc. As it is now, there's probably very few items that can be cooked by themselves, just things like [popcorn](https://github.com/vgstation-coders/vgstation13/blob/24d0c3327d1cc605340f8bdc8f19c897a3889a60/code/modules/food/recipes_microwave.dm#L806).

https://github.com/vgstation-coders/vgstation13/assets/145183032/ab56c880-825d-434d-9f2d-992a6e48a59b

### CODE REVIEW

**Comment by toomykins** [comment]
this is nice, but wasn't the intent to grill things rather than implement microwave recipes?

**Comment by falcon2346** [comment]
There are cooked meat items that don't require any other ingredients. If you cook meat over a fire on a spit you get rotisserie meat.

**Comment by ghost** [comment]
I'm a cheflet so I forgot about rotisserie/grills, just was thinking of microwaves and pans on the stovetop which also use microwave recipes. I should have perhaps looked at [`/obj/machinery/cooking/grill`](https://github.com/vgstation-coders/vgstation13/blob/24d0c3327d1cc605340f8bdc8f19c897a3889a60/code/modules/food/cooking_machines.dm#L655). But this is a good still a starting point. We could potentially have both where popcorn is makeable with a welding tool and meat can be grilled (without the matching recipe).

I can spend some time looking at `/obj/machinery/cooking/grill` if we want cooking with a welder and other things to basically do the same thing as that, and modify the new `/obj/item/weapon/reagent_containers/food/proc/cook` for it to do that.

**Comment by ghost** [comment]
Maybe in `/obj/item/weapon/reagent_containers/food/proc/cook` we can create a temporary grill and put it somewhere (inside the welding tool as toomy suggested?)

**Comment by pedr69** [comment]
>doesn't add carbon/burntmess/toxins into the cooked steak

**Comment by Hinaichigo** [comment]
Nice, this is a good direction to move in with the energy concept.
And if the reagents heating PR passes you could also then cook steak by exposing it to hot air.

**Comment by Blithering** [comment]
Since you can make meat into cutlets which cook straight off in a microwave with no further ingredient needs, would that be more useful as a starting point?

**Comment by Hinaichigo** [comment]
Maybe a simple way to do this without using recipes is, once the reagents inside the `food/snacks` item reaches a given temperature (the temperature could be item-specific), the item becomes 'roasted'. Upon roasting you could change the icon a bit, add some description about it being roasted, do some reagent changes like adding carbon or whatever, etc.  All the energy stuff will be handled for you under the hood by virtue of specific heat, and you can then roast apples, pizza, cheese wedges, etc. with minimal additional effort.
And that has the added bonus of being able to roast things via other methods or reagent heating.

**Comment by Lumbermancer** [comment]
I don't like this, like, at all. You should not be adding more features that make chef/kitchin even less relevant. You should be doing the opposite.

**Comment by wotisjanitor** [comment]
EVERYONE CAN DO EVERYTHING!!!!!

**Comment by ghost** [comment]
We never really determined if this was something people actually wanted in the issue, so I'll just leave this as a draft for a while and if the upvotes/downvotes indicate its controversial or unwanted I'll close it.

File: code/modules/food/recipes_microwave.dm [hunk 1]
```diff
@@ -813,7 +813,7 @@
 	result = /obj/item/weapon/reagent_containers/food/snacks/meatsteak/synth
 
 /datum/recipe/meatsteak
-	reagents = list(SODIUMCHLORIDE = 1, BLACKPEPPER = 1)
+	// reagents = list(SODIUMCHLORIDE = 1, BLACKPEPPER = 1)
```
Comment (Kurfursten) [inline code/modules/food/recipes_microwave.dm:816]: you can't just change an extremely commonly made recipe with no cl

Comment (ghost) [reply code/modules/food/recipes_microwave.dm:816]: That was only for testing purposes and wouldn't be in the finished PR.

**Comment by Kurfursten** [comment]
why is this not resulting in roasted food like the campfire? steak has salt and pepper

**Comment by ghost** [comment]
> why is this not resulting in roasted food like the campfire? steak has salt and pepper

Cause embarrassingly I totally forgot about grills, campfires, etc. and worked on this entirely microwave brained.

It's workable to have it work like roasting but I don't want to spend time on getting it working that way if the server doesn't actually want the feature.

**Comment by Hinaichigo** [comment]
I think people might be receptive to the idea of roasting things with heat, in a manner similar to grilling, if you approached it that way.
Also, maybe you could consider looking at making the frying pan cook things if you heat it enough with a welding tool.

**Comment by Hinaichigo** [comment]
Another thing...
Adding salt and pepper to meat directly using the shakers, and then cooking that in a pan, doesn't make meatsteak.
However, adding salt and pepper to the pan directly, and then cooking along with meat, does result in meatsteak.
I wonder if there's a nice way these could be reconciled.

**Comment by Kromkar** [comment]
Meme PRs should be closed, especially meme PRs coded by ChatGPT

**Comment by Lumbermancer** [comment]
do not bully

**Comment by Eneocho** [comment]
> I think people might be receptive to the idea of roasting things with heat, in a manner similar to grilling, if you approached it that way. Also, maybe you could consider looking at making the frying pan cook things if you heat it enough with a welding tool.

I'm telling you man, the ghetto cooktop you can make with a can and some rods.

**Comment by Hinaichigo** [comment]
That'd be cool, especially if you don't have access to wood for a campfire.

**Pull request closed without merging**